### ISSUE TITLE
[Feat]: Disable new UI

### ISSUE THREAD

**Issue by foxt** [netdata/netdata#15640]
### Problem

Hello. Is it possible to disable the new UI entirely? Currently you havbe to have click in the area where the old link is when linking and hope you clicked in the right spot and clicked fast enough.

### Description

Just give us a toggle to go back to the old UI.

There are also concerns about Netdata automatically installing propriatary code licensed under a new UI, and loading code from the Netdata servers (which may be changed at any time or monitored for analytics) (see also, https://github.com/netdata/netdata/issues/15466#issuecomment-1648074020)
The link for the license is only shown in a very small link for a brief period of time (only 1 second)

### Importance

really want

### Value proposition

The reason I ask is the new UI makes the assumption that you

1. are using multiple nodes with ND installed.
2. are using the cloud hosted service.

I am doing neither. I just have one node that i have Netdata installed and the old UI is much more conducive to this usecase.

For example, if you are not logged into the cloud, you cannot change any Netdata settings in the new UI.

![image](https://github.com/netdata/netdata/assets/12257112/28ef2c86-5516-4a9d-af2c-668225e1cf2c)

### Proposed implementation

Ideally, when Netdata made the migration, instead of pushing users to the new UI automatically, you'd have a box saying "we've got a new UI! would you like to try it out?" and you could say "yeah sure" or, "no, don't remind me"

and from both UI there'd be a settings toggle to switch between. I mean you'd have to actually implement a settings box in the new UI first but still.

**Comment by aldem** [comment]
I would suggest (if it is going to be accepted at all) to make it an option in the configuration file, not in the UI itself.

The problem with UI (all of them) than literally nothing could be configured in the configuration file - everyone gets the same set of defaults and all changes are only stored in a browser (or cloud) while not everyone is using one browser or allows cookies and not everyone is using cloud.

**Comment by hugovalente-pm** [comment]
thanks for sharing your feedback @foxt will ask some further question for us to try to understand from the old UI what  you're really missing

> Currently you havbe to have click in the area where the old link is when linking and hope you clicked in the right spot and clicked fast enough.

the old UI is still availabe, if you go to `/v1` you can get to old dashboard - like this: https://sanfrancisco.my-netdata.io/v1

> There are also concerns about Netdata automatically installing propriatary code licensed under a new UI, and loading code from the Netdata servers (which may be changed at any time or monitored for analytics) (see also, #15466 (comment))
The link for the license is only shown in a very small link for a brief period of time (only 1 second)

there was a big concern with the licensing around the new UI and all efforts were made to ensure we keep the Netdata Agent as open source but for sure we'll further investigate this concerns

regarding the link for the license you can find it here - https://github.com/netdata/netdata/blob/master/web/gui/v2/LICENSE.md

> Value proposition
>The reason I ask is the new UI makes the assumption that you
> 1. are using multiple nodes with ND installed.

could explain a bit why for the case of monitoring a single node the new UI doesn't meet your needs? is it the multi-instance charts (discussed on a discussion mentioned below) or something else?

> 2. are using the cloud hosted service.

the new UI is totally independent of you having a Cloud account or not, there are some features related to authorization/authentication (like Functions) or other related with ease of customization that link to a Cloud account but simple monitoring and troubleshooting like it was on the old UI should still be possible. if you find something amiss could you share with us? 

> For example, if you are not logged into the cloud, you cannot change any Netdata settings in the new UI.

it is true that access to the old UI settings is no longer available but most of them should be available on the new UI directly on the charts or other components. if there is a setting that you find missing do let us know and we'll try to see if can bring it into the specific components.

also there is this [GH discussion](https://github.com/netdata/netdata/discussions/15415) about this move to the new UI where we explain the rational for the move and also confirm that we need to work on an expanded view to give back instance charts (not sure this is what you mostly miss)

**Comment by aldem** [comment]
> the new UI is totally independent of you having a Cloud account or not
> ..
> it is true that access to the old UI settings is no longer available but most of them should be available on the new UI directly on the charts or other components.

This is not entirely true. The problem with new UI is that is has too many tunables and any change that you make in any chart (grouping, aggregation, chart type etc) is not preserved - once you close the browser or reload the page everything is lost.

Some settings (like metrics in "Nodes" tab) are not preserved across reload too, you get a message "Connect to cloud to be able to save your settings".

While v1 dashboard does not have many of those settings, it does show more relevant information by default even without customization.

Yes, v2 is very nice, very advanced and useful, it is just... not really user friendly by default in some parts, especially for those of us who is roaming between browsers too often, or who does not have dashboard open 24/7 (which is difficult because of enormous RAM usage if you leave it open & focused for a while).

> could you share what configurations you would like to see available in the configuration file?

- default dashboard's version
- timeframe
- for every chart (by context) - default grouping mode, aggregation and chart type, as existing defaults sometimes quite meaningless (like summing up latencies or displaying average across significantly different ranges)
- filters for events adn alerts (to show only "Errors" and "Critical" for instance)
- metrics displayed in "Nodes" tab

**Comment by andrewm4894** [comment]
> The problem with new UI is that is has too many tunables and any change that you make in any chart (grouping, aggregation, chart type etc) is not preserved - once you close the browser or reload the page everything is lost.

agree - just cross linking this feature request that might help: https://github.com/netdata/netdata-cloud/issues/789

**Comment by aldem** [comment]
@hugovalente-pm 

>  the configurations per chart (grouping, aggregation, chart type) is something new that you could couldn't manage on the old UI

Yes, I couldn't - but I don't need to as everything is visible, I just have to scroll (or click menu on the right). In v2 I have no choice and must click more times to choose what I need, every time - even after I switched to Events and then back to Overview, and this is really annoying.

By `timeframe` I mean the default period covered when I open the UI - last 10 minutes, last day etc.

Filters for events and alerts - yes, I mean default selected filters.

**Comment by hugovalente-pm** [comment]
> even after I switched to Events and then back to Overview, and this is really annoying.

This is a bug, we previously supported this but somehow on the process of the merge of these UIs it was broken - there is this bug to track it https://github.com/netdata/netdata-cloud/issues/903

> By timeframe I mean the default period covered when I open the UI - last 10 minutes, last day etc.
Filters for events and alerts - yes, I mean default selected filters.

These are some interesting visualization preferences to add, we don't even support these on Netdata Cloud. Made this feature request to track this https://github.com/netdata/netdata-cloud/issues/904

**Comment by KaKi87** [comment]
> the old UI is still available, if you go to `/v1` you can get to old dashboard

Is it still receiving bug fixes and security fixes, if yes for how long ?

Thanks

**Comment by hugovalente-pm** [comment]
@KaKi87 the old UI is held on `netdata/dashboard` if you check the [README there](https://github.com/netdata/dashboard/blob/master/README.md) it mentions this is deprecated

**Comment by KaKi87** [comment]
So the answer is no, so using it is not an answer to this issue.

**Comment by hugovalente-pm** [comment]
> So the answer is no, so using it is not an answer to this issue.

The path forward is improve and make needed changes on the new UI, in the meantime the old UI can be used but no work will be done on it

**Comment by KaKi87** [comment]
Well, the improvement we want is basically a KISS mode : Keep It Simple Stupid.

An uncomplicated mode.

Thanks

**Comment by pgassmann** [comment]
> The path forward is improve and make needed changes on the new UI, in the meantime the old UI can be used but no work will be done on it

This means that there is no longer a supported open source UI. Which makes your claim invalid. @ktsaou https://www.netdata.cloud/open-source/

Please communicate this clearly and inform the users about the proprietary frontend license. Or rather make the frontend open source!

**Comment by Forza-tng** [comment]
> > So the answer is no, so using it is not an answer to this issue.
> 
> The path forward is improve and make needed changes on the new UI, in the meantime the old UI can be used but no work will be done on it

How would you like to go about this? I think a lot of users have commented on the negative value to aggregate different metrics as one.

Some examples
IPMI monitoring combines fan speeds as one metric. This is counter-productive as faulty fans aren't visible. I simply can't see if one of 6 fans is running slower than usual. The same applies to temperatures and voltages.

In a similar way I cannot easily draw conclusions out of any of the container/cgroup charts as averages do not really give insight in the health of individual services. Even on a relatively small system, a service that goes rogue and pins one CPU would not be visible unless I am actively looking for that service by filtering the chart on the service.

In general, I'd say this is the major drawback of the new design. The possibility to have a quick glance to see out of order issues/stats is removed. Netdata now provides an overview, but few details, which is at odds with the enormous capacity to gather detailed statistics.

**Comment by hugovalente-pm** [comment]
> How would you like to go about this? I think a lot of users have commented on the negative value to aggregate different metrics as one.

We have two main tickets that relate to what we plan to work on this:
* https://github.com/netdata/netdata-cloud/issues/894 --> allow the expanding of charts based on some attribute, for example it could be the instance
* https://github.com/netdata/netdata-cloud/issues/789 --> allow the storing of chart settings

We are also working on improving the Custom Dashboards that will allow to make your own dashboards in an easier way

We weren't able to complete these because of other priorities that arose but we are discussing on trying to get to work on these beginning of this year

**Comment by Forza-tng** [comment]
> > How would you like to go about this? I think a lot of users have commented on the negative value to aggregate different metrics as one.
> 
> We have two main tickets that relate to what we plan to work on this:
> 
> * [[Feat]: ability to "expand by" or similar for charts (e.g. expand by container, mount point etc) netdata-cloud#894](https://github.com/netdata/netdata-cloud/issues/894) --> allow the expanding of charts based on some attribute, for example it could be the instance
> * [[Feat]: ability to save/override chart defaults in a room netdata-cloud#789](https://github.com/netdata/netdata-cloud/issues/789) --> allow the storing of chart settings

That sounds good. Perhaps something like a filter on the right side could filter all relevant charts. For example if i put cgroup: somegroup, it would filter all. cgroups to only include that one.

Still, i preferr having a tree view on the right side ao i can easilly select all objects individually.

> 
> We are also working on improving the Custom Dashboards that will allow to make your own dashboards in an easier way

This is good. But needs more effort from users.

> 
> We weren't able to complete these because of other priorities that arose but we are discussing on trying to get to work on these beginning of this year

Perhaps you can communicate a roadmap on the features you indend to implement, a this will ease users worry.

I hope you see how the current layout is not as usable as it was, otherwise it will be hard to take in our (users) needs.

**Comment by hugovalente-pm** [comment]
> That sounds good. Perhaps something like a filter on the right side could filter all relevant charts. For example if i put cgroup: somegroup, it would filter all. cgroups to only include that one.
> Still, i prefer having a tree view on the right side ao i can easilly select all objects individually.

that is an interesting suggestion. we had discussed of two things that relate to this:
* extending the Global Nodes filter to allow filter also with chart labels and could perhaps be extended to the _instances_, this would filter across all the charts under Metrics tab that had that attribute
* improve the current TOC search since it is still quite basic

we'll see this probably after what I mentioned above

> Perhaps you can communicate a roadmap on the features you indend to implement, I think this will ease users worry.

yes, we've been speaking internally of having some sort of public roadmap. let's see if we come around on releasing one

> I hope you see how the current layout is not as usable as it was, otherwise it will be hard to take in our (users) needs.

I believe this is debatable, we have also gotten very good feedback of all the efforts done around the new charts, the NIDL framework, etc.
we understand some use-cases were impacted by this, hence why we are discussing about finding away to improve them, but there is still a lot of value on what we delivered

**Comment by Forza-tng** [comment]
@hugovalente-pm 
> believe this is debatable, we have also gotten very good feedback of all the efforts done around the new charts, the NIDL framework, etc.
> we understand some use-cases were impacted by this, hence why we are discussing about finding away to improve them, but there is still a lot of value on what we delivered

Fair enough. The Netdata chart system and the overall functionality is really great, especially the very powerful collector and retention possibilities. I was speaking from my own use case as a home user, where I wanted to quickly be able to check the status of my server and vms without having to drill down to see details. Because I know my systems pretty well, and being able to see individual services as separate charts, allows me to quickly see if trends are going the wrong way. This worked very well up until the new UI came along, which was quite disappointing.

That said, I did not mean to sound so harsh before. I'd be happy to help work out some more detailed work flows that would suit my use cases. Perhaps that should be a discussion for Discord or IRC?

 🙏

**Comment by hugovalente-pm** [comment]
> I'd be happy to help work out some more detailed work flows that would suit my use cases. Perhaps that should be a discussion for Discord or IRC?

for sure, once we have done some initial brainstorm/prototyping from our side we'll let you know and get some feedback. 
thanks for the honest share and being available to helps us out figuring the best way on this case

**Comment by foxt** [comment]
Opened Netdata this morning and got a message that I can only assume was intended to scare me into upgrading to a Enterprise subscription, insinuating that without it my machines were insecure.


And with that, I give up and just reverted to the old version of Netdata Debian ships as part of their repos.

**Comment by hugovalente-pm** [comment]
> Opened Netdata this morning and got a message that I can only assume was intended to scare me into upgrading to a Enterprise subscription, insinuating that without it my machines were insecure.
> 
> And with that, I give up and just reverted to the old version of Netdata Debian ships as part of their repos.

thanks for the feedback @foxt, the goal isn't to scare but to let people know how the Netdata Agent works by default. we provide information on our documentation for how you can [secure your nodes](https://learn.netdata.cloud/docs/configuring/securing-netdata-agents/) by: Expose Netdata only in a private LAN, Use an authenticating web server in proxy mode, Use Netdata parents as Web Application Firewalls, etc.

all of the above require you to do these setups yourself, the Netdata Enterprise Agent is an offering we are exploring and see if people would be interested in such a thing were we would take some burden out of your hands

**Comment by foxt** [comment]
> the goal isn't to scare but to let people know how the Netdata Agent works by default. we provide information on our documentation for how you can

The CTA wasn't 'read the docs', it was 'upgrade to Enterprise'. The link was not to the docs for what I have, it was to your pricing page.

**Comment by hugovalente-pm** [comment]
> The CTA wasn't 'read the docs', it was 'upgrade to Enterprise'. The link was not to the docs for what I have, it was to your pricing page.

Yes, we know the link wasn't to the docs. We are trying to measure the interest of such an offering that would replace the work a user would need to do by himself.
The docs are there for many time back, users interesting on securing their agents will probably have gone through those documents.

**Comment by foxt** [comment]
> The docs are there for many time back, users interesting on securing their agents will probably have gone through those documents.

And those not interested, or those who have already read the docs, don't need a nagbar telling them to upgrade to Enterprise

**Comment by hugovalente-pm** [comment]
>  I'd be happy to help work out some more detailed work flows that would suit my use cases. Perhaps that should be a discussion for Discord or IRC?

@Forza-tng are you still up on this offer? we would like to better discuss your use-case and expectations?

**Comment by KaKi87** [comment]
Well, I ended up removing Netdata from my servers.

The new UI isn't only overcomplicated, but also was causing high iowait.

As I basically was using the app as a web `htop`, I'll replace it with [Glances via homepage](https://gethomepage.dev/latest/widgets/services/glances/) instead.

Bye

**Comment by Forza-tng** [comment]
@hugovalente-pm 
> > I'd be happy to help work out some more detailed work flows that would suit my use cases. Perhaps that should be a discussion for Discord or IRC?
> 
> @Forza-tng are you still up on this offer? we would like to better discuss your use-case and expectations?

Hi, sorry for the late reply. Yes I would be interested in this :)

**Comment by tkatsoulas** [comment]
> Well, I ended up removing Netdata from my servers.
> 
> The new UI isn't only overcomplicated, but also was causing high iowait.
> 
> As I basically was using the app as a web `htop`, I'll replace it with [Glances via homepage](https://gethomepage.dev/latest/widgets/services/glances/) instead.
> 
> Bye

I really don't get it, even if the new UI is overcomplicated (I would say enriched with tons of functionality, but as user myself when things change, it can bring me some inconvenience) we didn't stop shipping the old UI. Access it via your `localhost:19999/v1`. 

You want it default? overwrite the   `/usr/share/netdata/web/index.html` with `/usr/share/netdata/web/v1/index.html` *

BEST: solution add a reverse proxy to point to v1 dashboard. Guys make it work for your case this an open-source project and needs to evolve for all the users. 

`*` you need to do that in every update

**Comment by KaKi87** [comment]
> we didn't stop shipping the old UI

But you stopped maintaining it.

**Comment by pgassmann** [comment]
@tkatsoulas 
> Guys make it work for your case this is an open-source project and needs to evolve for all the users.

The new v2 UI has a [proprietary license](https://github.com/netdata/netdata/blob/master/src/web/gui/v2/LICENSE.md), it's not open source. 

**Please put it under an open source license if you claim to be an open source project** or stop claiming netdata is an open source project if the only maintained UI is propietary!

**Comment by aldem** [comment]
I have the impression that all new developments are slowly rendering the v2 dashboard practically unusable without cloud integration.

Unless there is an option to configure locally (per instance) and persistently (not just during the browser session) for every chart, it will be painful to use.

For instance, defaulting to averaging all fan speeds or ping times is simply nonsensical, and reverting to defaults on every page reload is even more so - yet, that's how it is, with no signs of changing anytime soon. I do not want to choose relevant metrics every time again and again when I load the dashboard - I want it to be kept as I set it last time - _without cloud_ and even when I open the dashboard in different browser.

In v1 (pre-v2) most metrics are displayed in more convenient way, there is rarely need to change something, but in v2 this is different - it is like to install and tune new OS every time when you open your laptop.

What is worse, all the changes introduced for v2, while they have their advantages, are making v1 and v0 less usable with each new release.

Even simple request - an option to show v1 by default, barely an hour of work - completely ignored or has so "low priority" that we will never see it.

All of this really feels like slowly boiling a frog - at some point, it will be so inconvenient to use Netdata in its default setup without cloud integration that every user will be forced to either use the cloud or choose another solution.

Most ironic, though, is the use of the killer feature - functions. It feels so strange that you have to subscribe to the cloud (free or not - it doesn't matter, it's still the cloud) for "privacy reasons." What? Do I have to register with a third party somewhere and share some private data (at least my IP address) just to use something that is generated locally? How is this "privacy"?

I have no problem when a project is leaning in the direction of "we need to make [more] money," but I have an issue when this is done covertly and non-transparently. Unfortunately, that's how it feels now, and I have no doubts that I am not alone in this sentiment.

Yes, I know, there is a free tier (and probably always will be) - but this is not an option because it is cloud. However small the amount (and nature) of data is sent somewhere - there are _no guarantees_ that it will be properly protected and will not be used for anything else. Besides, it won't work at all in closed networks - and I am not talking about commercial usage, even home lab may be totally airgapped (and there is no free Netdata On-Prem).

**Comment by tkatsoulas** [comment]
Guys thank you so much for taking the time to express your thoughts/concerns/opinions, rest assured, I'll take them down after careful consideration and transfer them to the rest of the company (who are not following this thread).

**Comment by amfern** [comment]
I don't like the v2 because there is no metrics explanation, like it was for v0 and v1.

![image](https://github.com/user-attachments/assets/3ec7f7d3-825f-4d78-b43a-617235948fac)

**Comment by ralphm** [comment]
@amfern This information is still there, but as we added many controls for slicing and dicing, to reduce the required vertical space per chart, we moved it to the info icon:

![image](https://github.com/user-attachments/assets/ad17c6fc-86c0-4f90-a295-a0b6e0480630)
![image](https://github.com/user-attachments/assets/b74b0df9-9be7-4792-be21-9730992510ea)

**Comment by CreeperFace00** [comment]
I have a fairly extensive parent/child setup with about 40 nodes streaming metrics, bear in mind this is entirely for my homelab and not any commercial use. 
I've been using the old v1 interface since the new one came out because I really did not like it, I figured I'd check it out today because I though "surely it's gotten better, right?" only to be met with this:

![Image](https://github.com/user-attachments/assets/ca9b1f20-65c2-4c06-8ea7-b37e2ec683ee)
This is just so heartbreaking to see because I have been using Netdata practically daily for years and have constantly told people how amazing it is. Worse yet, I checked some of my newer nodes only to find that the v1 dashboard is not shipped anymore. 
This is all just so disappointing and unfortunate. I do not want to be forced to use the cloud and pay monthly for something I am hosting on prem as an individual. 
Eventually I am sure the v1 dashboard will be broken or flat out removed from old installs, at which point I'll have to revert to a backup and disabled updates.

**Comment by ralphm** [comment]
@CreeperFace00 thank you for using Netdata! We changed the way we license the UI to ensure continued development, by requiring a paid plan for showing charts for more than 5 nodes in the dashboard. The local dashboard for each node is still available as always.

We then more recently removed the older versions of the dashboard from the Agent because they had been unmaintained for a long time, broken in several ways because of changes we made to how metrics are collected, and to have a single UI shared between the Agent local dashboard and the one used in Cloud.

For homelab use we have a dedicated plan with a flat fee of $90 per year or $10 per month, instead of a price per node as with the business plan. With this, you can continue to use the local dashboard, but it does require a token "dance" by logging into Netdata Cloud, to verify being on the plan.

**Comment by aldem** [comment]
@ralphm 

> they had been unmaintained for a long time

Lets face the inconvenient truth - not "had been", but because the NetData team stopped maintaining it. It did not happen by itself, it was a deliberate decision.

v2/v3 dashboards are often inconvenient to use without the cloud, and the cloud is not free - even for the free plan, the data is leaving the premises (yes, "metadata" is also data); even worse - it is tunneled through Internet, which means - increased traffic, increased latency, no real guarantees about privacy (regardless of what you claim in documentation) etc.

Everything was done to make v2/v3 dashboards inconvenient without the cloud and to force people to enter the cloud (nag screens & reminders, concealing access without cloud), even local (per-node) preferences (changes to charts etc) cannot be stored in the local agent while being a trivial thing to implement, and this is not going to change (all such requests are ignored).

I personally would gladly pay $99 or even $199 (per year) for a home lab plan, but _only_ if not a _single bit_ of (meta)data will leave my premises and it should work totally without Internet connectivity, with local storage (not in the browser!) of changes made to the dashboard - I don't need all that advanced analytics and probably 99% of other stuff provided by the cloud. And I am quite sure I am not alone, but who cares?

Now, let me guess...  NetData v3  (or even v2.7) will be, most likely, not accessible locally at all. You will be required to subscribe... at least it feels like this.

Too bad, as NetData is really amazing, just... heavily commercialized. And _not really_ open-source anymore.

**Comment by aleks-mariusz** [comment]
> can copy v1 stuff

I'm open to what this looks like, I don't really know my way around netdata hacking unfortunately.. what needs to be moved?

> eventually v1 will be broken

It seems to already have been removed _entirely_ (as per my screenshot) on versions _newer_ than 2.2.6 (i had to run dnf downgrade _repeatedly_, until curl stopped returning a 404 on http://host:19999/v1/)

> still works with 2.5

that's a relief, but seems to fly in the face of the netdata org's claim that v1 "is no longer compatible with the Netdata Agent API" (despite having earlier _also_ claimed they would not remove v1).  But in recent times I'm getting the impression the netdata org cares more about profits than being true to their words; shame as this was one of the most amazing monitoring projects way back in the day before the whole cloud-integration being shoved down our throats.
